In [4]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [7]:
df = pd.read_csv("../dataset/ab_data.csv")
df.head()

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

In [9]:
df.shape

(294478, 5)

In [10]:
df['user_id'].nunique()

290584

In [11]:
df[df['converted']==1].shape[0]/df.shape[0]

0.11965919355605512

In [12]:
df[((df["group"]== "treatment")== (df['landing_page']=="new_page")) ==False].shape[0]

3893

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 294478 entries, 0 to 294477
Data columns (total 5 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   user_id       294478 non-null  int64 
 1   timestamp     294478 non-null  object
 2   group         294478 non-null  object
 3   landing_page  294478 non-null  object
 4   converted     294478 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 11.2+ MB


In [17]:
# check if there are any missing value
df.isna().sum()

user_id         0
timestamp       0
group           0
landing_page    0
converted       0
dtype: int64

In [18]:
#check if there is dublicate value

df.duplicated().sum()

np.int64(0)

In [19]:
df2 = df[((df["group"]== "treatment")== (df['landing_page']=="new_page")) ==True]
df2.head()

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

In [20]:
df2[((df2["group"]== "treatment")== (df2['landing_page']=="new_page")) ==False].shape[0]

0

In [22]:
# check how many users id are unique
df2['user_id'].nunique()

290584

In [23]:
df2[df2.duplicated(['user_id'])]

user_id                   timestamp      group landing_page  converted
2893   773192  2017-01-14 02:55:59.590927  treatment     new_page          0

In [24]:
df2.drop_duplicates(['user_id'], keep='first', inplace=True)
df2.shape

C:\Users\user\AppData\Local\Temp\ipykernel_16192\4069054304.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df2.drop_duplicates(['user_id'], keep='first', inplace=True)


(290584, 5)

In [25]:
df2.head()

user_id                   timestamp      group landing_page  converted
0   851104  2017-01-21 22:11:48.556739    control     old_page          0
1   804228  2017-01-12 08:01:45.159739    control     old_page          0
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0
4   864975  2017-01-21 01:52:26.210827    control     old_page          1

In [26]:
df2['converted'].mean()

np.float64(0.11959708724499628)

In [32]:
#Given that an individual was in the control group, what is the probability they converted?
df_control = df2[df2['group'] == 'control']
num_users = df_control.shape[0]
num_convert = df_control[df_control['converted']==1].shape[0]
proba_convert_control = num_convert/num_users
proba_convert_control

0.1203863045004612

In [33]:
#given that an individual was in the treatment group, what is the probability they converted?
df_treat = df2[df2['group'] == 'treatment']
num_users = df_treat.shape[0]
num_convert = df_treat[df_treat['converted']==1].shape[0]
proba_convert_treat = num_convert/num_users
proba_convert_treat

0.11880806551510564

In [34]:
#  What is the probability that an individual received the new page?

df2[df2['landing_page']== 'new_page'].shape[0]/df2.shape[0]

0.5000619442226688

In [35]:
df2.groupby(['group', 'converted']).size()

group      converted
control    0            127785
           1             17489
treatment  0            128046
           1             17264
dtype: int64

Consider your results from each parts above, and explain below whether you think there is sufficient evidence to conclude that the new treatment page leads to more conversions.

No, there is no sufficient evidence that the new page has higher conversion factors than that of old page ,the number of customers in the control and treatment groups are nearly equal, 12% of control group customer have been converted and 11.88% of treatment(new_page) group have converted, in general the old_page conversion rate is slightly higher than that for new_page but additional test required to confirm this